plat/drivers/gic/gic-v3.c

```diff
@@ -606,3 +606,43 @@ int gicv3_probe(const void *fdt, struct _gic_dev **dev)
 	return 0;
 
 }
+
+/**
+ * Send a software generated interrupt to the specified core.
+ * 
+ * @sgintid the software generated interrupt id
+ * @cpuid the id of the targeted cpu
+ */
+void gic_sgi_gen(uint8_t sgintid, uint32_t cpuid)
+{
+	uint64_t sgi_register = 0, control_register_rss, type_register_rss;
+	uint64_t range_selector = 0, aff0;
+
+	/* Only INTID 0-15 allocated to sgi */
+    UK_ASSERT(sgintid <= GICD_SGI_MAX_INITID);
+    sgi_register |= (sgintid << 24);
+
+	/* Set affinity fields and optional range selector */
+	sgi_register |= (cpuid & MPIDR_AFF3_MASK) << 16;
```

Comment:
Shift bits are not in accordance with affinity positions in ICC_SGI_ELn register: 
https://developer.arm.com/documentation/ddi0595/2020-12/AArch64-Registers/ICC-SGI1R-EL1--Interrupt-Controller-Software-Generated-Interrupt-Group-1-Register

Reply:
Done, sorry for this, I don't see how I missed it.